### PR TITLE
chore: update css variables info

### PR DESCRIPTION
**Pull request by tomzemp**
This removes the `(not recommended)` parenthetical about using <CssVariables> component for making DHIS2 UI constants accessible in CSS.

Summary of points from slack conversation where we discussed why we originally had this language:
- ...the original rationale behind that was that these constants are used in the design system and that apps are made using the design system so they should not be needed. (Hendrik)
- Maybe there was a time when we were using css-in-js styles that could consume the constants from the UI library? (Kai)

In summary, we now recognize that it is common to need these values (e.g. colors) for styling and that as we mostly use css modules for styling, it makes sense to not officially discourage use of <CSSVariables> component.

### CODE REVIEW

**Review by KaiVandivier — APPROVED** [review]
This could be a `docs: ` commit 🙂 I'm wondering if that will be enough to trigger a redeployment of the docsite though 🤔 (probably, I think -- if not, it probably won't be long until a `fix:` commit is pushed)
